jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPartFormData.java

```diff
@@ -78,6 +78,52 @@ private MultiPartFormData()
     {
     }
 
+    /**
+     * Returns {@code multipart/form-data} parts using the given {@link Content.Source} and {@link MultiPartConfig}.
+     *
+     * @param content the source of the multipart content.
+     * @param attributes the attributes where the futureParts are tracked.
+     * @param contentType the value of the {@link HttpHeader#CONTENT_TYPE} header.
+     * @param config the multipart configuration.
+     * @return the future parts
+     */
+    public static CompletableFuture<MultiPartFormData.Parts> from(Content.Source content, Attributes attributes, String contentType, MultiPartConfig config)
+    {
+        // Look for an existing future (we use the future here rather than the parts as it can remember any failure).
+        CompletableFuture<MultiPartFormData.Parts> futureParts = MultiPartFormData.get(attributes);
+        if (futureParts == null)
+        {
+            // No existing parts, so we need to try to read them ourselves
+
+            // Are we the right content type to produce our own parts?
+            if (contentType == null || !MimeTypes.Type.MULTIPART_FORM_DATA.is(HttpField.getValueParameters(contentType, null)))
+                return CompletableFuture.failedFuture(new IllegalStateException("Not multipart Content-Type"));
+
+            // Do we have a boundary?
+            String boundary = MultiPart.extractBoundary(contentType);
+            if (boundary == null)
+                return CompletableFuture.failedFuture(new IllegalStateException("No multipart boundary parameter in Content-Type"));
+
+            // Look for an existing future MultiPartFormData.Parts
+            futureParts = MultiPartFormData.from(attributes, config.getMultiPartCompliance(), config.getViolationListener(), boundary, parser ->
```

Comment:
Can we avoid using this strange form of the `from` method.  Ultimately I'd like to deprecate it and remove all the setters from the parser and just use the config and it's builder.

Reply:
I agree with @gregw

Reply:
I have now used the `from` taking `MultiPartConfig` for the servlet implementation and deprecated the other `from` methods.

Reply:
but you are still using the from here that takes a Consumer.  Can you inline and simplify that from here.

Reply:
Ok, I rewrote this to not use the deprecated `from` method.